src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -60,13 +60,20 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        $response = $event->getResponse();
+
         if ($session->isStarted() || ($session instanceof Session && $session->hasBeenStarted())) {
-            $event->getResponse()
-                ->setPrivate()
-                ->setMaxAge(0)
-                ->headers->addCacheControlDirective('must-revalidate');
+            if (!$response->headers->has('Symfony-Session-NoAutoCacheControl')) {
```

Comment:
imho this should be a public constant. then the constant can also be documented a little bit to explain what this is about.